lib/Core/Executor.cpp

```diff
@@ -3003,6 +3007,27 @@ void Executor::callExternalFunction(ExecutionState &state,
     return;
   }
 
+#ifndef WINDOWS
+#ifndef __APPLE__
+  /* From /usr/include/errno.h: it [errno] is a per-thread variable. */
+  int *errno_addr = __errno_location();
+#else
+  int *errno_addr = __error();
+#endif
+  // Update errno value explicitly
+  ObjectPair result;
+  bool resolved = state.addressSpace.resolveOne(
+      ConstantExpr::create((uint64_t)errno_addr, Expr::Int64), result);
+  if (!resolved)
+    klee_error("Could not resolve memory object for errno");
+  int error = externalDispatcher->getLastErrno();
+  if (memcmp(result.second->concreteStore, &error, result.first->size) != 0) {
```

Comment:
Why can't this be handled with copyInConcretes()? I don't particularly like that concreteStore is becoming public just for this.

Reply:
Yeah, in #838 we are trying to use that as default for everything, so it should be easier next time. But how is it relevant to this issue?